src/server/session.ts

```diff
@@ -2737,6 +2737,15 @@ namespace ts.server {
                 // TODO: report errors
                 return this.requiredResponse(/*response*/ true);
             },
+            [CommandNames.UpdateFileSystem]: (request: protocol.UpdateFileSystemRequest) => {
+                this.changeSeq++;
+                this.projectService.updateFileSystem(
+                    request.arguments.created && arrayIterator(request.arguments.created),

```

Comment:
Why are we turning the arrays into iterators?

Reply:
(It looks like you're following a precedent, so I'm probably missing something.)

Reply:
The precedent is applyChangesInOpenFiles, which turns out to be the only one to use iterators. I *think* it's because it wants to cheaply and transparently iterate over some arrays backward, which is not a concern here. I'll simplify to arrays.